### PR TITLE
[PyTorch][Inference] PT2.3 Inference Currency for x86 cpu/gpu ec2/sm

### PR DESCRIPTION
**Pull request by sallyseok**
### Description
- Added an unconditional skip to `test_serving.py::test_serve_csv` and `test_serving.py::test_serve_cpu_model_on_gpu` because they were showing issues, and I've concluded that they are test flaws regarding distinction between cpu and gpu. test_serving.py needs to be reevaluated and fixed/replaced with newer tests/models for torchserve. In the meantime, I removed DataParallel from `test/sagemaker_tests/pytorch/inference/resources/mnist/model_cpu/1d/code/mnist_1d.py` because 1. DataParallel is no longer recommended by pytorch and 2. it is supposed to be a cpu model. To account for this, I also replaced the model tar file with a new model.pth that is compatible with a network not using DataParallel. 

### Tests run
- All tests for sagemaker images passing on [68952eb](https://github.com/aws/deep-learning-containers/pull/3982/commits/68952ebd9319b0c5ce94f53d2850a62eb75d9ced)
- All tests for ec2 passing on [cc0b83f](https://github.com/aws/deep-learning-containers/pull/3982/commits/cc0b83ff40a8be24abe9318a8c9c1ddf810bf733)

**NOTE: By default, docker builds are disabled. In order to build your container, please update dlc_developer_config.toml and specify the framework to build in "build_frameworks"**
- [x] I have run builds/tests on commit <INSERT COMMIT ID> for my changes.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**
<details>
<summary>Expand</summary>

- [ ] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

</details>

### Formatting
- [x] I have run `black -l 100` on my code (formatting tool: https://black.readthedocs.io/en/stable/getting_started.html)

#### Builds to Execute
<details>
<summary>Expand</summary>

Fill out the template and click the checkbox of the builds you'd like to execute

*Note: Replace with <X.Y> with the major.minor framework version (i.e. 2.2) you would like to start.*

- [ ] build_pytorch_training_<X.Y>_sm
- [ ] build_pytorch_training_<X.Y>_ec2

- [x] build_pytorch_inference_2.3_sm
- [x] build_pytorch_inference_2.3_ec2
- [ ] build_pytorch_inference_<X.Y>_graviton

- [ ] build_tensorflow_training_<X.Y>_sm
- [ ] build_tensorflow_training_<X.Y>_ec2

- [ ] build_tensorflow_inference_<X.Y>_sm
- [ ] build_tensorflow_inference_<X.Y>_ec2
- [ ] build_tensorflow_inference_<X.Y>_graviton
</details>

### PR Checklist 
<details>
<summary>Expand</summary>

- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

#### NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ec2_benchmark_tests = true` or `sagemaker_benchmark_tests = true`
</details>

### Pytest Marker Checklist
<details>
<summary>Expand</summary>

- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type
</details>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/2.3/py3/cu121/Dockerfile.gpu [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+
+FROM ec2 AS sagemaker
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+LABEL com.amazonaws.sagemaker.capabilities.multi-models=true
+
+###
+LABEL com.amazonaws.sagemaker.inference.cuda.version=12.1
```
Comment (roywei) [inline pytorch/inference/docker/2.3/py3/cu121/Dockerfile.gpu:302]: it should be com.amazonaws.sagemaker.inference.cuda.verified_versions

File: pytorch/inference/docker/2.3/py3/cu121/Dockerfile.gpu [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+#                      |___/                        |_|
+#################################################################
+
+FROM ec2 AS sagemaker
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+LABEL com.amazonaws.sagemaker.capabilities.multi-models=true
+LABEL com.amazonaws.sagemaker.inference.cuda.verified_versions=12.2
```
Comment (arjkesh) [inline pytorch/inference/docker/2.3/py3/cu121/Dockerfile.gpu:300]: Does this label need to match our cuda version in the container? If so, can you add a test in https://github.com/aws/deep-learning-containers/blob/master/test/dlc_tests/sanity/test_dlc_labels.py to verify values?

Comment (sallyseok) [reply pytorch/inference/docker/2.3/py3/cu121/Dockerfile.gpu:300]: No, this number is not the same as the actual cuda version in the container. The naming is a bit misleading haha

File: dlc_developer_config.toml [hunk 3]
```diff
@@ -69,7 +69,7 @@ ec2_benchmark_tests = false
 ec2_tests_on_heavy_instances = false
 
 ### SM specific tests
-### Off by default
+### On by default
```
Comment (sallyseok) [inline dlc_developer_config.toml:72]: Fixing the comment here since this was changed to true by default

**Review by roywei — APPROVED** [review]

**Pull request merged**